### PR TITLE
Remove lib/devicetrust from lib/client

### PR DESCRIPTION
**Pull request by rosstimothy**
The native device trust libraries included in lib/devicetrust prevent building client tools that don't use device trust from building with cgo disabled. This moves the default device auth ceremony and device enrollment from lib/client into a location specific to tsh since it's the only tool that requires them.

Updates https://github.com/gravitational/teleport/issues/43112.

### CODE REVIEW

File: lib/client/api.go [hunk 1]
```diff
@@ -3388,7 +3384,7 @@ func (tc *TeleportClient) DeviceLogin(ctx context.Context, rootAuthClient authcl
 	// Allow tests to override the default authn function.
 	runCeremony := tc.DTAuthnRunCeremony
 	if runCeremony == nil {
-		runCeremony = dtauthn.NewCeremony().Run
```
Comment (codingllama) [inline lib/client/api.go:3391]: Can this cause problems to Connect?

Same below.

**Review by codingllama — APPROVED** [review]
LGTM.

Thanks for all the testing you did to make sure this is safe!

File: lib/client/api.go [hunk 2]
```diff
@@ -463,18 +461,16 @@ type Config struct {
...
-	// Defaults to "dtauthn.NewCeremony().Run()".
 	DTAuthnRunCeremony DTAuthnRunCeremonyFunc
 
 	// dtAttemptLoginIgnorePing and dtAutoEnrollIgnorePing allow Device Trust
 	// tests to ignore Ping responses.
 	// Useful to force flows that only typically happen on Teleport Enterprise.
 	dtAttemptLoginIgnorePing, dtAutoEnrollIgnorePing bool
 
-	// dtAutoEnroll allows tests to override the default device auto-enroll
+	// DTAutoEnroll allows tests to override the default device auto-enroll
```
Comment (gzdunek) [inline lib/client/api.go:471]: This comment is no longer true, correct? We set this value in production code, not only in tests and we don't have the default value.
If yes, we should also update `DTAuthnRunCeremony`.

**Review by gzdunek — APPROVED** [review]
Connect changes LGTM.

However, I didn't test it, if I'm correct, it requires a regular tag build (if you have one, I can the test device trust in Connect).

**Comment by codingllama** [comment]
> However, I didn't test it, if I'm correct, it requires a regular tag build (if you have one, I can the test device trust in Connect).

@gzdunek you can get a local build going yourself, or one of us can send you a pre-built binary. I rarely do tag builds for dev only purposes.

**Comment by gzdunek** [comment]
Thanks @codingllama. I tested device trust in Connect, works fine.

**Pull request merged**